app/src/main/java/com/practicum/playlistmarket/player/data/repository/PlayerRepositoryImpl.kt

```diff
@@ -4,44 +4,37 @@ import android.media.MediaPlayer
 import android.os.Handler
 import android.os.Looper
 import android.util.Log
-import com.practicum.playlistmarket.player.domain.api.TrackStateListener
-import com.practicum.playlistmarket.player.domain.api.TrackTimeListener
+
 import com.practicum.playlistmarket.player.domain.repository.PlayerRepository
 import com.practicum.playlistmarket.player.domain.StatePlayer
+import com.practicum.playlistmarket.player.domain.api.PlayerListener
 import java.text.SimpleDateFormat
 import java.util.*
 
-class PlayerRepositoryImpl(
-    private var trackTimeListener: TrackTimeListener,
-    private val trackStateListener: TrackStateListener,
-) :
-    PlayerRepository {
+class PlayerRepositoryImpl : PlayerRepository {
+
+    var listener : PlayerListener? = null
```

Comment:
```suggestion
    private var listener : PlayerListener? = null
```
так как у тебя есть отдельная функция `setupListener`